### PR TITLE
Implemented: Available to promise base implementation

### PR DESCRIPTION
**Pull request by ravilodhi**
Implemented: Added entities for building rules for setting threshold, safety stock, allow pickup and allow brokering values. Also added seed data for the same.

### CODE REVIEW

File: data/AtpSeedData.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<entity-facade-xml type="ext-seed">
+    <org.apache.ofbiz.common.enum.EnumerationType enumTypeId="AVAILABLE_TO_PROMISE" description="Available to promise"/>
```
Comment (dixitdeepak) [inline data/AtpSeedData.xml:4]: We should use the Moqui entities

File: entity/DecisionRuleEntities.xml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            <key-map field-name="conditionTypeEnumId" related="enumId"/>
+        </relationship>
+        <relationship type="one" fk-name="RULE_COND_RULE" related="co.hotwax.common.DecisionRule" short-alias="decisionRule">
+            <key-map field-name="ruleId"/>
+        </relationship>
+        <master>
+            <detail relationship="ruleConditionType"/>
+        </master>
+    </entity>
+    <entity entity-name="RuleAction" package="co.hotwax.common" group="ofbiz_transactional" short-alias="ruleAction">
```
Comment (dixitdeepak) [inline entity/DecisionRuleEntities.xml:109]: we can use co.hotwax.rule as package name

File: drl/ProductFacilityRuleTemplate.drl.ftl [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<#assign decisionRules = ec.entity.find("co.hotwax.common.DecisionRule").condition("ruleGroupId", ruleGroupId).condition("statusId", "ATP_RULE_ACTIVE").orderBy("sequenceNum asc").list()!>
+<#if decisionRules?has_content>
+
+  <#-- Set the package and imports -->
+  package co.hotwax.product;
```
Comment (dixitdeepak) [inline drl/ProductFacilityRuleTemplate.drl.ftl:5]: co.hotwax.rule

File: entity/DecisionRuleEntities.xml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<entities xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://moqui.org/xsd/entity-definition-3.xsd">
+    <entity entity-name="RuleGroup" package="co.hotwax.common" group="ofbiz_transactional" short-alias="ruleGroup">
```
Comment (dixitdeepak) [inline entity/DecisionRuleEntities.xml:22]: We can use co.hotwax.rule as package name

File: service/co/hotwax/common/DecisionRuleServices.xml [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        <out-parameters><parameter name="recordsRemoved" type="Long"/></out-parameters>
+        <actions>
+            <script>
+                import org.moqui.context.ExecutionContext
+                import org.moqui.entity.EntityCondition
+                ExecutionContext ec = context.ec
+                Calendar basisCal = ec.user.getCalendarSafe()
+                basisCal.add(Calendar.DAY_OF_YEAR, (int) - daysToKeep)
+                basisTimestamp = new Timestamp(basisCal.getTimeInMillis())
+                recordsRemoved = ec.entity.find("moqui.service.job.ServiceJobRun")
```
Comment (dixitdeepak) [inline service/co/hotwax/common/DecisionRuleServices.xml:16]: we should not use the deleteAll, as for bulk data it fails, we should delete the records using iterator

File: data/AtpSeedData.xml (hunk 1 above)
Comment (ravilodhi) [reply data/AtpSeedData.xml]: Done. Thanks!

File: entity/DecisionRuleEntities.xml (hunk 2 above)
Comment (ravilodhi) [reply entity/DecisionRuleEntities.xml]: Done. Thanks!

File: drl/ProductFacilityRuleTemplate.drl.ftl (hunk 3 above)
Comment (ravilodhi) [reply drl/ProductFacilityRuleTemplate.drl.ftl]: Done. Thanks!

File: entity/DecisionRuleEntities.xml (hunk 4 above)
Comment (ravilodhi) [reply entity/DecisionRuleEntities.xml]: Done. Thanks!

File: service/co/hotwax/common/DecisionRuleServices.xml (hunk 5 above)
Comment (ravilodhi) [reply service/co/hotwax/common/DecisionRuleServices.xml]: Done. Thanks!

**Pull request merged**